staging/src/k8s.io/api/resource/v1beta1/types.go

```diff
@@ -142,6 +142,63 @@ type ResourceSliceSpec struct {
 	// +optional
 	// +listType=atomic
 	Devices []Device `json:"devices" protobuf:"bytes,6,name=devices"`
+
+	// PerDeviceNodeSelection defines whether the access from nodes to
+	// resources in the pool is set on the ResourceSlice level or on each
+	// device. If it is set to true, every device defined the ResourceSlice
+	// must specify this individually.
+	//
+	// Exactly one of NodeName, NodeSelector, AllNodes, and PerDeviceNodeSelection
+	// must be set.
+	//
+	// +optional
+	// +oneOf=NodeSelection
+	// +featureGate=DRAPartitionableDevices
+	PerDeviceNodeSelection *bool `json:"perDeviceNodeSelection,omitempty" protobuf:"bytes,7,name=perDeviceNodeSelection"`
```

Comment:
If we do v1beta2 we might change `AllNodes` and `NodeName` to pointers too - the lack of consistency is weird.

Reply:
Created https://github.com/kubernetes/kubernetes/issues/130932 to track the API change proposal related with v1beta2 :)